docs/rules/terraform_module_shallow_clone.md

```diff
@@ -0,0 +1,71 @@
+# terraform_module_shallow_clone
+
+Require pinned Git-hosted Terraform modules to use shallow cloning.
+
+## Example
+
+```hcl
```

Comment:
Tighten this up to one `module "git"` block highlighting the violating this rule detects. Every variation creates too much visual noise.